snap/info.go

```diff
@@ -1425,6 +1426,26 @@ func (app *AppInfo) SecurityTag() string {
 // DesktopFile returns the path to the installed optional desktop file for the
 // application.
 func (app *AppInfo) DesktopFile() string {
+	desktopFileIDs, err := app.Snap.DesktopPlugFileIDs()
+	if err != nil || len(desktopFileIDs) == 0 {
+		// fallback to a simple heuristic "$PREFIX_$APP.desktop"
+		return filepath.Join(dirs.SnapDesktopFilesDir, fmt.Sprintf("%s_%s.desktop", app.Snap.DesktopPrefix(), app.Name))
+	}
+	// Loop through desktop-file-ids desktop interface plug attribute in order to
+	// have deterministic output
+	for _, desktopFileID := range desktopFileIDs {
+		desktopFile := filepath.Join(dirs.SnapDesktopFilesDir, desktopFileID+".desktop")
+		if !osutil.FileExists(desktopFile) {
+			continue
+		}
+		// No need to also check instance name because we already filter by the
+		// snap's desktop file ids
+		de, err := desktopentry.Read(desktopFile)
+		if err == nil && de.SnapAppName == app.Name {
```

Comment:
is it intentional we ignore errors here? maybe a comment here why we ignore error

Reply:
Yes, errors when reading indicates either an issue opening the desktop file or a malformed desktop file, both of which are internal errors caused somewhere else.

The original behavior of this function was best effort where it guessed the desktop file for the app, the intention was to keep the spirit of best effort when extending it to support desktop file ids as well.

Good point, I will add a comment, and also log the error for debugging.

Reply:
Added comment and added logging